app/src/main/java/com/github/se/eduverse/viewmodel/FileViewModel.kt

```diff
@@ -71,4 +77,54 @@ class FileViewModel(val fileRepository: FileRepository) {
     _validNewFile.value = false
     _newFile.value = null
   }
+
+  /**
+   * Open a pdf file using an specialized app on the device
+   *
+   * @param fileId the id of the file as stored in firestore
+   * @param context the context in which this method is used
+   */
+  fun openFile(fileId: String, context: Context) {
+    fileRepository.accessFile(
+        fileId = fileId,
+        onSuccess = { pdfRef ->
+          val localFile = File.createTempFile("tempFile", ".pdf")
+          pdfRef
+              .getFile(localFile)
+              .addOnSuccessListener { openPDF(localFile, context) }
+              .addOnFailureListener {
+                Log.e("Open File", "Opening of file ${pdfRef.name} failed: $it")
+                Toast.makeText(context, "Can't open file", Toast.LENGTH_SHORT).show()
+              }
+        },
+        onFailure = {
+          Log.e("Access File", "Access of file at $fileId failed: $it")
+          Toast.makeText(context, "Can't access file", Toast.LENGTH_SHORT).show()
+        })
+  }
+
+  /**
+   * Open a pdf file. Parameters intent and uri are there to make the function testable
+   *
+   * @param file the file to open
+   * @param context the context in which the file is opened
+   * @param intent the intent that is created and then executed with an appropriate activity
+   * @param uri the uri of the file in specified context
+   */
+  fun openPDF(
+      file: File,
+      context: Context,
+      intent: Intent = Intent(Intent.ACTION_VIEW),
+      uri: Uri =
+          FileProvider.getUriForFile(context, "${BuildConfig.APPLICATION_ID}.fileprovider", file)
+  ) {
```

Comment:
You added Intent and Uri as optional parameters, which is a good idea for testing. However, in most cases you could avoid recreating a new Intent or Uri if the operation has already failed or succeeded previously.